### PR TITLE
containers add Queue and testing

### PR DESCRIPTION
**Pull request by dagar**
Simple intrusive Queue container in the style of the existing List. Used in the new PX4 work queue. https://github.com/PX4/Firmware/pull/11570

This is intentionally not implemented as a List as to allow placing things in both containers simultaneously.

### CODE REVIEW

**Comment by dagar** [comment]
Updated to make everything in the node type private and named verbosely to potentially minimize future pain mixing with other containers.

File: src/systemcmds/tests/test_IntrusiveQueue.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+		delete node;
+
+		ut_compare("size check", q1.size(), 100 - i - 1);
+	}
+
+	// verify list has been cleared
+	ut_assert_true(q1.empty());
+	ut_assert_true(q1.size() == 0);
+
```
Comment (bkueng) [inline src/systemcmds/tests/test_IntrusiveQueue.cpp]: I'd add another `pop()` here

File: src/include/containers/IntrusiveQueue.hpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+	void push(T newNode)
+	{
+		if (newNode->next_intrusive_queue_node() != nullptr) {
+			// error, node already queued
+			return;
+
+		} else {
+			// pushed node's next is empty,
+			//  check that it's not already the head or tail
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/include/containers/IntrusiveQueue.hpp]: Done.

File: src/include/containers/IntrusiveQueue.hpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+			} else {
+				// only one item left
+				_head = nullptr;
+				_tail = nullptr;
+			}
+		}
+
+		// clear next in popped (in might be re-inserted later)
+		ret->set_next_intrusive_queue_node(nullptr);
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/include/containers/IntrusiveQueue.hpp]: Thanks, moved into `!empty()` above.

File: src/systemcmds/tests/test_IntrusiveQueue.cpp (hunk 1 above)
Comment (dagar) [reply src/systemcmds/tests/test_IntrusiveQueue.cpp]: Good idea, done.

**Pull request merged**